### ISSUE TITLE
Element cannot be closed / shutdown cleanly

### ISSUE THREAD

**Issue by OR13** [transmute-industries/sidetree.js#420]
In order for jest to exit properly, element needs to close all its activity properly.

This is not happening leading to the default jest warning:

```
Jest did not exit one second after the test run has completed.

This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.
```

**Comment by thomas-tran** [comment]
I got the same issues when implement on azure database ledger since the connection of the database is still open and there is no method from the did method core to close the connection or terminate the ledger for the shutdown method

**Comment by OR13** [comment]
yes, I have struggled to exit the processes cleanly... perhaps explicit `process.exit(0)` is needed.

**Comment by OR13** [comment]
can you link to your source?

**Comment by thomas-tran** [comment]
In the Azure SQL database ledger implement, I can close the database connection after the test script https://github.com/sekuid/sidetree.js/blob/main/packages/ledger-ms-sqldb/src/__tests__/AzureSqlDbLedger.test.ts

However, in the did method , I could not do it as there is no method to terminate the ledger properly as the database connection pool is still open. This is not an issue with the actual system as the backend service is support to reuse the connection from the connection pool. However, it will cause an issue during testing or shutting down the system.
https://github.com/sekuid/sidetree.js/blob/main/packages/did-method-atom/src/test/wallet.test.ts

Note: The did-method-atom is used Azure SQL database with ledger enabled (the test will use Ms SQL Server 2022) and Azure storage as the Content address storage  (test will used azurite as a storage emulator) 

https://github.com/sekuid/sidetree.js/blob/main/docker-compose.yml

**Comment by OR13** [comment]
`did-method-atom` super cool!

So i am hearing we ned to expose a shutdown handler from the did method core package, is that right?

Something in here: https://github.com/transmute-industries/sidetree.js/blob/main/packages/did-method/src/Core.ts#L162

**Comment by thomas-tran** [comment]
Yes, any function in the IBlockchain interface to terminate/shutdown the ledger e.g
```
 await this.versionManager.stop();

 await this.blockchain.stop();
```
